tellerlogger/src/test/kotlin/com/kiliansteenman/teller/logger/data/room/MapConverterTest.kt

```diff
@@ -0,0 +1,104 @@
+package com.kiliansteenman.teller.logger.data.room
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+internal class MapConverterTest {
+
+    private val mapConverter = MapConverter()
+
+    private companion object {
+        private const val KEY_VALUE_SPLIT = "<==>"
+        private const val PARAM_SPLIT = "<;;>"
+    }
+
+    @Nested
+    inner class ToMapTest {
+
+        @Test
+        fun `when value is empty, then empty map is returned`() {
+            assertEquals(0, mapConverter.toMap("").size)
+        }
+
+        @Test
+        fun `when value is blank, then empty map is returned`() {
+            assertEquals(0, mapConverter.toMap(" ").size)
+        }
+
+        @Test
+        fun `when value is just a separator, then empty map is returned`() {
+            assertEquals(0, mapConverter.toMap(PARAM_SPLIT).size)
+        }
+
+        @Test
+        fun `when the string contains a single key value pair, then map contains a single item`() {
+            val map = mapConverter.toMap("key${KEY_VALUE_SPLIT}value")
```

Comment:
What happens when I emit `key` here? Like : `"${KEY_VALUE_SPLIT}value"`. Do we need a test case for this?